services/scan/src/precinct_scanner_app.ts

```diff
@@ -275,41 +296,29 @@ export async function buildPrecinctScannerApp(
       return;
     }
 
-    if (getBallotsCounted(store) > 0) {
-      response.status(400).json({
-        status: 'error',
-        errors: [
-          {
-            type: 'ballots-cast',
-            message:
-              'cannot change the precinct selection if ballots have been cast',
-          },
-        ],
-      });
-      return;
-    }
-
-    store.setPrecinctSelection(bodyParseResult.ok().precinctSelection);
-    store.setPollsState('polls_closed_initial');
-    await updateInterpreterConfig(interpreter, workspace);
+    store.setIsSoundMuted(bodyParseResult.ok().isSoundMuted);
     response.json({ status: 'ok' });
   });
 
-  app.get<NoParams, Scan.GetPollsStateConfigResponse>(
-    '/precinct-scanner/config/polls',
+  app.get<NoParams, Scan.GetPrecinctScannerElectionSessionResponse>(
+    '/precinct-scanner/electionSession',
     (_request, response) => {
-      const pollsState = store.getPollsState();
-      response.json({ status: 'ok', pollsState });
+      response.json({
+        isTestMode: store.getTestMode(),
+        pollsState: store.getPollsState(),
+        ballotCountWhenBallotBagLastReplaced:
+          store.getBallotCountWhenBallotBagLastReplaced(),
+      });
```

Comment:
I like how you've packaged up the election session!

The abstraction defined between app and store feels a bit murky to me. Who manages the rules about election sessions, the app or the store? Currently, the app assembles the election session and decides what changes merit resetting it, but the store knows what resetting a session means. The downside of this is that the logic for maintaining the integrity of the session is slightly spread out, which makes it more likely we'll mess it up at some point.

A few directions I could imagine moving in:
1. The store doesn't know anything about election sessions and only puts data into slots in the election table on the db
2. The store offers CRUD methods for election sessions, the app decides when election sessions get created/reset based on other changes (e.g. changing config)
3. Option 2, plus the store enforces rules about when sessions are created/reset

There are tradeoffs to each approach. I'm not really sure what the vision for the "store" is, and would be curious to hear @eventualbuddha's thoughts as well.

Without thinking deeply about it, I'd probably pick option 2, which would treat election session as it's own data type (even though secretly the store is storing it in the election table).

Reply:
@jonahkagan thanks for taking a look at this! I definitely can see the murkiness you’re talking about. I think Option 1 wouldn’t actually prevent the sort of footguns i think you’re worried about. Option 2 could, but it seems to me like it might be over investing in this new concept of the election session. 

The election session is created initially from that point on is only ever reset. If the app were just calling for the reset, and the store were managing the reset, that would be a fine division of function. But the murkiness is where the app is also packaging the election session for the frontend API. I think this is a tolerable amount of murkiness for now, since for now these concepts are just buckets of data that we’re separating for readability rather than use case.

Reply:
Well, there's a bit more murkiness than that. The store has methods to get/set individual parts of the election session as well. So you could conceivably reset one without changing the others if you didn't see the specific reset method. I think things start to get dangerous when there are multiple ways to do things.

I agree with you that the logic isn't super complicated today, but establishing a clear foundation with defined abstraction lines between different modules will help it stay clear as it develops over time.

Reply:
The app has to be able to get/set fields of the election session somehow, right? Do you just want a clearer API, like store.electionSession.method()?

I think we have a good indication from product that this is not an abstraction that will be built upon. I don’t foresee having multiple election sessions. Or doing anything other than resetting them. It’s just a matter deciding what is reset when switching modes and not. Perhaps myopic, sure.

Moreover, I think we’re thinking a little too much about this. It doesn’t strike me as very important. I’d like to hear from @eventualbuddha and move on. I’m happy with a maximalist approach, where election sessions are packaged and distinguished in the store and even separated in the DB on a separate table. I’m happy with the current approach. I just want to push back against investing time into abstractions that I don’t see being foundational. I’ll defer to y’all. 

I’m curious about your thoughts on packaging things together for the API between frontend in backend, as I’ve drafted here. It made sense to me to make 1 or 2 requests rather than however many pieces of data we have.

Reply:
If you think it's not a useful abstraction, that's totally fine with me. I would say in that case just remove it. I'm mainly pushing back against having a "halfway" abstraction.

Reply:
I don't think that the abstraction is useful in that it's not helpful for future work that we can anticipate. The goal was to introduce clarity, in response to your comment [here](https://github.com/votingworks/vxsuite/pull/2751#discussion_r1007352267). But it sounds like this "halfway" abstraction creates as many additional potential issues as it does clarity.

In that case, I will remove it, and just leave all "operating state" like polls state and ballot bag last replaced count behind the `/config` endpoint.